### PR TITLE
Modify channel range assertion

### PR DESCRIPTION
**Pull request by markccchiang**
This code modification is according to @pford's suggestion (issue #265).

### CODE REVIEW

File: Region/Region.cc [hunk 1]
```diff
@@ -422,7 +422,11 @@ bool Region::MakeExtensionBox(casacore::WCBox& extend_box, int stokes, ChannelRa
         if (channel_range.to == ALL_CHANNELS) { // extend to nchan
             max_chan = all_channels;
         }
-        assert(max_chan >= min_chan && all_channels >= max_chan >= 0 && all_channels >= min_chan >= 0);
+        assert((max_chan >= min_chan) && (all_channels >= max_chan));
```
Comment (pford) [inline Region/Region.cc]: I am confused by the code block before this assertion.  If the image has 64 channels (all_channels=64) and the channel_range is (ALL_CHANNELS, ALL_CHANNELS), shouldn't min_chan=0 and max_chan=all_channels-1?  It is setting the range to (64, 64) which would be invalid.  But it looks like the ALL_CHANNELS range is never used, Frame sets the ChannelRange argument to GetRegion to a single channel, a range of channels, or ChannelRange() (which does the right thing).

Comment (pford) [reply Region/Region.cc]: therefore (all_channels > max_chan), not >=

**Comment by markccchiang** [comment]
I think I should also correct `max_chan = all_channels` as `max_chan = all_channels - 1`.

File: Region/Region.cc (hunk 1 above)
Comment (markccchiang) [reply Region/Region.cc]: Sorry for the confusion! For `ChannelRange()` sets the channel range (0, ALL_CHANNELS - 1). For ChannelRange(ALL_CHANNELS, ALL_CHANNELS) also sets the channel range (0, ALL_CHANNELS - 1). Is it looks reasonable?

**Pull request merged**